src/main/java/pl/patrycja/ox/gameflow/Game.java

```diff
@@ -2,31 +2,45 @@
 
 import pl.patrycja.ox.GameSettings;
 import pl.patrycja.ox.Sign;
-import pl.patrycja.ox.winnerchecker.Spectator;
-import pl.patrycja.ox.winnerchecker.SpectatorsRoom;
+import pl.patrycja.ox.ui.UI;
 
 import java.util.List;
 
-class Game {
+class Game extends Mode {
 
-    private GameSettings gameSettings;
+    public Game(UI ui) {
+        super(ui);
+    }
 
-    private Player firstPlayer = new Player("A", Sign.CROSS);
-    private Player secondPlayer = new Player("B", Sign.NAUGHT);
+    @Override
+    public List<Player> createPlayers() {
+        ui.display("Please enter name for first player.");
+        Player firstPlayer = new Player(ui.read(), Sign.CROSS);
+        ui.display("Please enter name for second player.");
+        Player secondPlayer = new Player(ui.read(), Sign.NAUGHT);
+        return List.of(firstPlayer, secondPlayer);
+    }
 
-    Game(GameSettings gameSettings) {
-        this.gameSettings = gameSettings;
+    @Override
+    public void settings(String[] inputArrayParameters) {
+        gameSettings = setup(inputArrayParameters);
     }
 
-    void play() {
-        gameSettings.setPlayer();
-        PlayerChanger playerChanger = new PlayerChanger(List.of(firstPlayer, secondPlayer), gameSettings);
+    private GameSettings setup(String[] inputArrayParameters) {
+        checkCorrectInputData(inputArrayParameters);
+        return updateValuesForBoardAndUnbrokenLine(inputArrayParameters);
+    }
 
-        for (int i = 0; i < gameSettings.getNumberOfMatches(); i++) {
-            List<Spectator> spectators = SpectatorsRoom.addSpectators(gameSettings);
-            Match.init(gameSettings, spectators)
-                    .addController(playerChanger)
-                    .start();
+    private GameSettings updateValuesForBoardAndUnbrokenLine(String[] inputArrayParameters) {
+        GameSettings.GameSettingsBuilder gameSettings = GameSettings.builder();
+        int firstParameter = Integer.parseInt(inputArrayParameters[0]);
```

Comment:
Is that board size, maybe worth to name it like that?

Reply:
Yes, you are right.